### ISSUE TITLE
New Rule: Sentences per line

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [eslint/markdown#317]
### Rule details

Enforce that each line of text has at most one sentence

### What type of rule is this?

Warns about a potential problem

### Example code

```markdown
See spot. See spot run.
```

### Participation

- [x] I am willing to submit a pull request to implement this rule.

### Additional comments

I'd previously made https://github.com/JoshuaKGoldberg/sentences-per-line as a contributed rule for Markdownlint. IMO keeping to one sentence per line is a good practice. Copying from its README: sentences-per-line makes sure no line contains more than one sentence. This is useful because:

* Shorter lines result in simpler, easier-to-understand Git diffs
* Longer lines are harder to read in source code

We can define a "sentence" here as any characters followed by a text punctuation. `Abc.` would be a sentence; `Abc. Def.` and `Abc! Def?` would be two sentences. 

I'd understand if this is too opinionated for `@eslint/markdown` core - but do want to propose at least providing it as a rule.

**Comment by nzakas** [comment]
> IMO keeping to one sentence per line is a good practice

It's funny, I have the exact opposite opinion on this. :smile:

Is this something that Prettier can enforce?

**Comment by JoshuaKGoldberg** [comment]
They chose not to in https://github.com/prettier/prettier/issues/3302. The difficulty there was determining what constitutes a "sentence": https://github.com/prettier/prettier/issues/3302#issuecomment-346891816.

This could be done as a Prettier plugin though, which I personally would be satisfied with. 🙂

**Comment by nzakas** [comment]
Yeah, I have the same concern over what constitutes a sentence. I think a Prettier plugin is probably a better way to go.
